### ISSUE TITLE
Wan 2.1 t2v (Text to Video) LoRA training - TIPS needed 🙏

### ISSUE THREAD

**Issue by AlonDan** [kohya-ss/musubi-tuner#275]
Hi everyone,

I'm VERY new to training with Musubi-Tuner, 
I need YOUR tips & tricks for my very first t2v (Text to Video) LoRA training.

My very first LoRA training was for i2v (Image to Video) focused on MOTION, and it was successfully based on some tests I did, but it was just a simple test nothing I would actually use, I'm still learning and trying to understand some properties and options for fine tuning my future LoRA trainings.

---

This time,
I would like to train a specific **CHARACTER** using i2v, but I'm already guessing it's different than training MOTION in many ways.
I'll be happy to read your Tips & Tricks based on YOUR experience and if you got Good / Bad results because of... some changes I can learn from, please share.

I'll start with some questions:

**1** - From my understanding, VIDEO dataset is always better than still images, but that's based on reading in reddit and other sources.
Is it true?

**2** - If I will train the Dataset ONLY on IMAGES, what resolution should I use? 
Should I think like "FLUX" training: different angles, lights, positions, face expressions, etc.. or since it's TEXT to VIDEO the rules should be different in some ways?

**3** - Should I MIX both: IMAGES + VIDEOS ?
If so, what about different resolution sizes, different ratios, etc.. what is the BEST way to prepare the dataset so it won't ruin the potential of a GOOD training result ?

**4** - What if I will combine IMAGES + VIDEOS ?
How it's done without doing a mess in the dataset: should I place ALL Videos + Images in the SAME Folder?
Should I separate them to: dataset\Video and dataset\Image for example? and update the .TOML with extra: `[[dataset]]`
Can somebody give a .TOML example of how it should look?

**5** - TRAINING Settings / Properties:
What can you recommend me to do, please share a good example of something for such case and if possible EXPLAIN why you use specific properties or values.

**6** - Dataset amount:
How many IMAGES should I use for Dataset? (or videos? or both?) How many is NOT too many or too little? for t2v with good results!
It doesn't have to be THE BEST results, but something that will work nicely and above, I mean without low quality or inaccurate poor results.

**7** - ANY tips & tricks related to training a t2v (Text to Video) will help, also PLEASE share .TOML and Training Data if you already trained successfully and you are happy with the results, it will be nice to have reference START POINT for me to try on my first Text to Video character training.

Also,
If you have EXAMPLE t2v LoRA I can try in ComfyUI and you have the training properties used it will be VERY nice to examine and learn from it.
I guess Civiai is one way to look, but not only the properties always shared but it's also HUGE amount of LoRA's to look for.

---

If it helps, these are my specs:  
I still can't tell how to USE all of these in the training so any tips will be nice as long as it's safe to work

> Intel Core Ultra 9 285K
> Nvidia RTX 5090
> 96GB Ram
> Windows 11

Thanks ahead for your help! 🙏

---

BTW - Earlier I posted a **[GUIDE](https://github.com/kohya-ss/musubi-tuner/issues/273)** on how I installed it for RTX-50xx in case you're interested.
It came with easy **3** Batch files to cache latents, train, convert your trained LoRA to ComfyUI.

**Comment by Sarania** [comment]
Please see https://github.com/kohya-ss/musubi-tuner/discussions/182 and also please maybe consider posting more social posts like this in discussions in the future 🙂 Oh also a good guide(linked in the thread too) is in the writeup for this model https://civitai.com/models/1404755/studio-ghibli-style-wan21-t2v-14b Cheers!

**Comment by AlonDan** [comment]
Thanks, I did look (and still do to compare information from experiments by other users) but it's not really answer most of my questions above related to the IMAGES / VIDEOS, combo etc.. I have some specific questions.

Most of the discussion is VERY advanced, people already know what they're talking about it's hard for a beginner to understand most of the properties, also like I said they don't really cover my specific questions in that thread.

Any help will be appreciated, also may help others who are a bit confused with the docs and also it can't harm to learn from others based on their experience and experimental, please share anything from what mentioned on my post.

Thanks ahead! 🙏

**Comment by Sarania** [comment]
These are the best answers I can give you:

1. Depends on what you're trying to achieve. For motion yes this is true. But for training characters, images are often both sufficient and the only option.

2. Personally I use 1280x720 when training images for Wan as that's it's native res. As far as choice, the same rules apply. If there's a part of a character you don't show (like the back) the model will have trouble envisioning that. In my experience, these models are more tolerant of lower quality than Flux, but quality of the dataset still matters a lot. 

3. As I understand, Wan was pretrained on images and then video came later. You can definitely mix them together training, for instance if you wanna show high res images for detail but lower res video for motion. There's no way to answer what is best as it depends on a WIDE amount of factors including what data you simply have available to you. These models don't need to see super HD video to train, so generally we stick to lower res like 480x272 or even 256x256 when training motion.

4. I've never done both at the same time nor do I use TOML so I'll pass on this one.

5. There's no one right set of training settings that work for every dataset and situation. These are the settings for my best Wan LoRA:
```
Date: 2025-04-16T10:42:43 Title: WanKisses
Resolution: 1280x720 Architecture: wan2.1/lora
Network Dim/Rank: 16.0 Alpha: 16.0 dtype: F16
Module: networks.lora_wan : {'loraplus_lr_ratio': '4'}
Learning Rate (LR): 2e-05
Optimizer: came_pytorch.CAME.CAME(weight_decay=0.01,eps=(1e-30, 1e-16),betas=(0.9, 0.999, 0.9999))
Scheduler: constant_with_warmup Warmup steps: 100
Epoch: 38 Batches per epoch: 64 Gradient accumulation steps: 1
Timestep sampling: Shift Discrete Flow Shift 3.0
```

why? Because a CRAP TON of experimentation and failed runs led me to that. But they won't be the best for your model unless you're training the exact same thing. It can be a useful starting point but it really does depend heavily and for instance you have very new, nice hardware that a lot of us don't have so what you can do is not the same as what we can do. The one thing that might warrant explaining beyond that  is "loraplus" but doing it without getting complex is... LoRA has multiple types of weights inside of it. Normally they are all trained at the same rate. LoRAPlus allows setting a different rate for the "A" blocks versus "B" and this produces better results that converge quicker. It's a multiplier and 4 is a good value for both Wan and Hunyuan. It means the "B" blocks train at 4 times the rate of the "A" blocks, which from my own testing and others, seems to be a good rate.

6. I tend to shoot for 20 images/videos for my first run of a single concept dataset, but some people use as little as 10. Somewhere around that number. 

7. See #5

**Comment by Sarania** [comment]
I edited my post a lot to add more info, sorry about that, it should be finalized now. One last thing I do wanna say is training models is definitely as much of an art as a science. You can get some basic guidelines and parameters to lay a groundwork of knowledge, but then it really does come down to some amount of experimentation. Both of my successful LoRA that I published to Civit... I'd say I've done 30 total training runs of EACH of those. In fairness, I started way early when we didn't know as much, but even now it usually takes several tries to get it right. I start with the basic settings but... some of it is intuition and feel that come from experience. If you have history training Flux, most of that knowledge is transferable here. I think Wan is pretty easy to train compared to Flux or even Hunyuan.

You have epic hardware so you can definitely use higher resolution/longer video/ etc than many of us! Or do it in half the time XD So you will get there!

**Comment by AlonDan** [comment]
Thanks @Sarania this is VERY helpful, I will look more into your example and others while I'm studying as I go.

I'm still trying to understand many things which are still very confusing to me and I can't really tell what to select per training goal yet, first it was i2v for motions / actions focus (I did learn something from reading a lot and train twice), 
now I'm trying to do a text 2 video for Characters for example, I'm sure actual training is the best way to learn but since it takes hours on every experiment for decent results, I'm trying to learn as much as I can from others experience first.

In example: learning rate, Network Dim, Optimizer, etc.. these are still very confusing to me, properties and values wise.

I'm digging on the DOCS, also CoPilot helps with some examples, but with all the respect, 
feedback such as yours is much more appreciated and helpful in my case, I appreciate it🙏

**Comment by Sarania** [comment]
You want to know what those do specifically?

Learning rate is as straightforward as it sounds. The basic training process works like this: Using your caption, a sample image is created with the model + LoRA and this is compared to your provided image/video. The difference is seen as "loss" and then the model weights are updated to try to "learn" this loss. This is an oversimplification but I think that's what you're after. Higher values make those numbers go up quicker - you will learn more quickly, but you may miss fine details/optimal convergence. Imagine that like running so fast you can't stop where you meant, you overshoot.

Network dimension is the actual size of your LoRA - the number of parameters it has. It's not an exact number, rather the exact number depends on which model the LoRA is for combined with the dimension. But basically larger - can learn more stuff, but might learn TOO much stuff and overpower other LoRA, vs smaller - can learn less, but plays nicer with other models and hopefully learns only what you want. I find 16 is a good value for Wan/Hunyuan and I never go above the default 32. These are BIG models so that's plenty.

Optimizer is a bit trickier to explain in simple terms. It's the algorithm/code controlling the gradient descent process during training. Gradient descent is complex stuff, but basically it's how the model makes a best guess as which direction to move it's weights in to try to learn the thing. Some optimizers are made for speed, others accuracy, others memory efficiency. For training video we mostly use memory efficient ones like the default AdamW8Bit (the 8 bit version of AdamW optimizer), or CAME Pytorch like I mentioned. Generally you wanna just find one that fits your needs and use it. Really the default is fine, don't worry about messing with this one.

**Comment by AlonDan** [comment]
@Sarania THIS IS PURE GOLD! thank you so much 🙏

I'm going to read it more carefully (again) when I'll start train my next LoRA, 
Even consider it's crazy late and I'm tired, your detailed explanations seems easier for me to follow, I'll come back to this thread during my next training experiments. thank you for taking the time to explain all it. 💙

**Comment by Sarania** [comment]
You're very welcome I'm glad it's helpful! Oh another tip I can give you MONITOR your training! Musubi provides a few ways, personally I use tensorboard. You'll want to make sure it's installed in your venv (activate it, then `pip install tensorboard`) and then add to your training command:
```
--log_with tensorboard --logging_dir /path/to/log/output
```

and then once training is started, activate your venv from a terminal and do:
```
tensorboard --logdir=/path/to/log/output
```

and this will give you a browser interface at http://localhost:6006/ where you can monitor training stats. Now as far as interpreting it that's a dark art, but the main thing is to watch average loss and it should be going down over time. You can watch the number on the command line but this gives you a nice set of graphs to look at. I've got a run going right now, here's my current tensorboard so you can see what that's like:

![Image](https://github.com/user-attachments/assets/535d87e4-9f0c-46bd-b77b-9271be181999)

Importantly, you can also compare between runs and see how various settings are affecting the overall training process in a more visual way. After you've seen a few, you'll get a feel for it. I also HIGHLY recommend creating at least still image samples during training. It tells you more than you think about how the process is going, even for motion LoRA.

**Comment by AlonDan** [comment]
NICE! this is another subject I was wondering about, since I'm a visual learner this should help!

Sadly, I can run **TensorBoard** (installed perfectly fine) but for some reason it doesn't show anything when training is running,
I tried to run it before training and during... the browser shows **TensorBoard** I even tried change to different option-tabs in it and even refresh the page... it seems like it doesn't "look" at Musubi-Tuner.

I installed it with the requirements as part of the Musubi-Tuner extra (optional) in the same environment, so I didn't do anything different.

I never used TensorBoard before, I've seen it few times when I played with Google Colab some years ago but nothing more than that, so I wonder if this issue I mentioned is known or easy to fix, I tried to fix it with ChatGPT for an hour or so but than I gave up on it...

I believe it's IMPORTANT to see this while training, it sure can help.

Like I mentioned, it's too late for my focus but I will sure come back to this thread MANY times, including the TensorBoard part if I'll ever be able to make it "look" at Musubi-Tuner while training.

Once again, thank you so much THIS thread is already pure gold I believe many newbies such as myself should bookmark this one thanks to your very detailed help! 🙏

**Comment by Sarania** [comment]
So if it's running but not showing anything, it's not picking up your logs. The path in both the training command and when you invoke tensorboard must be exactly the same. So maybe try specifying the full path both times:
```
--log_with tensorboard --logging_dir C:\Users\<your user name>\training\logs
```
and
```
tensorboard --logdir=C:\Users\<your user name>\training\logs
```

The "tensorboard" command should be run after Musubi has started. Also, it won't show anything until Musubi actually starts writing out datapoints since you've not used it before(eventually it will show your past training runs when you launch if you don't delete the logs) so wait until you hit like step 5 and hit refresh in your browser then. That should do the trick.
